eng/templates/stages/mirror.yml

```diff
@@ -48,17 +52,49 @@ stages:
       value: $(PIPELINE.WORKSPACE)/${{ parameters.dotnetStagingPipelineResource }}/$(packagesArtifactName)/$(shippingPackages)
     - name: releaseTag
       value: $[ stageDependencies.PreRelease.PreRelease.outputs['ReadReleaseInfo.ReleaseTag'] ]
+    - name: sdkVersion
+      value: $[ stageDependencies.PreRelease.PreRelease.outputs['ReadReleaseInfo.SdkVersion'] ]
+    - name: runtimeVersion
+      value: $[ stageDependencies.PreRelease.PreRelease.outputs['ReadReleaseInfo.RuntimeVersion'] ]
+    - name: releaseChannel
+      value: $[ stageDependencies.PreRelease.PreRelease.outputs['ReadReleaseInfo.ReleaseChannel'] ]
+    - name: combinedSmokeTestPrereqsTarballPath
+      value: $(Pipeline.Workspace)/dotnet-smoke-test-prereqs-$(sdkVersion).tar.gz
+
+    - ${{ if eq(parameters.dotnetMajorVersion, '6.0') }}:
+      - name: smokeTestPreReqsArtifactNameX64
+        value: Build_Tarball_x64 CentOSStream9-Offline_Artifacts
```

Comment:
This is going to create a point of maintenance throughout the lifecycle of updating/retiring the distros/versions our CI runs on.  What is the failure experience if this leg no longer exists?  Because the main CI pipeline is defined elsewhere there is no easy way to keep these in sync.

Reply:
If it didn't exist, you'd get an error like `Artifact <name> was not found for build <id>`. If you run a dry run ahead of time (which I do), you'll catch this at that time.

Other options:
* Update the VMR/tarball pipeline to have an additional stage that consolidates the smoke test prereqs from the build legs and outputs them to a well-known artifact that can be consume from the release pipeline.
* Update the VMR/tarball pipeline to publish well-known artifact names specifically for x64 and arm64 from the individual build legs (no additional stage). For example, `CentOSStream9_Offline_MsftSdk_x64` would publish a `SmokeTestPrereqs-x64` artifact and `Debian11_Offline_MsftSdk_arm64` would publish a `SmokeTestPrereqs-arm64` artifact. The release pipeline would consume these as hard-coded artifacts but with more stable names than the build legs. The key is to configure the pipeline so that one and only one build leg for each architecture is the one that publishes the prereqs artifact. That adds some maintenance burden in case of removal/addition of build legs.

From a simplicity standpoint, I consider what I've currently implemented to be the simplest. Yes, it can be broken but the churn on build legs isn't likely to be high and can be caught with dry run. And of course, the other options can cause breakage too if not maintained correctly.

Let me know your thoughts.

Reply:
I have no objections to the proposal.  The CI legs change infrequently and the error message is pretty clear.